### PR TITLE
Task7

### PR DESCRIPTION
**Pull request by Darina007**

### CODE REVIEW

File: task3/UI/script_task7.js [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+            this._adList.push(adItem);
+            return true;
+        }
+        return false;
+    }
+
+    addAll(adList) {
+        let notValidPosts = [];
+        adList.forEach((post) => {
+            if (AdCollection._validate(post) && this.get(post.id)) {
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: I think here should be `if (AdCollection._validate(post) && !this.get(post.id)) {`

File: task3/UI/script_task7.js [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+                        }
+                    )
+                    return flag;
+                })
+            }
+            if (filterConfig.validateUntil) {
+                workingArray = workingArray.filter(value => value.validateUntil - filterConfig.validateUntil >= 0)
+            }
+            if (filterConfig.discount) {
+                workingArray = workingArray.filter(value => value.discount - filterConfig.discount >= 0)
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: Why not just `value.discount >= filterConfig.discount`? The same for lines 57 and 64

File: task3/UI/script_task7.js [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+        workingArray = workingArray.slice(countSkippedPosts, countSkippedPosts + countReceivedPosts);
+        return workingArray;
+    }
+
+    get(id) {
+        return this._adList.find((item) => item.id === id);
+    }
+
+    edit(id, adItem) {
+        let mutablePost = this.get(id);
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: It's safer to copy object first (using `Object.assign` or object destructuring), only after that change it`s fields.

File: task3/UI/script_task7.js [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+        if (mutablePost) {
+            mutableFields.forEach((field) => {
+                    if (field !== 'id' && field !== 'createdAt' && field !== 'author') {
+                        mutablePost[field] = adItem[field];
+                    }
+                }
+            )
+            if (AdCollection._validate(mutablePost) && this.get(adItem.id)) {
+                this.remove(id);
+                this.add(mutablePost);
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: `Array.splice`

File: task3/UI/script_task7.js [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+        }
+        return flag;
+    }
+
+    remove(id) {
+        return this._adList.splice(this._adList.findIndex(item => item.id === id), 1);
+    }
+
+    clearAll() {
+        this._adList.splice(0, this._adList.length);
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: This is equivalent of just `this._adList = []` 😉

**Review by YauheniyaBasaranovich — APPROVED** [review]
👍  I accept eslint part of task as fully done, so your score is 11 now. You can increase it up to 12 by fixing my comments.

File: task3/UI/script_task7.js (hunk 1 above)
Comment (Darina007) [reply task3/UI/script_task7.js]: Thank you! 
I fixed all thing that you commented, can i merge this task?

File: task3/UI/script_task7.js [hunk 6]
```diff
@@ -75,7 +75,7 @@ class AdCollection {
     }
 
     edit(id, adItem) {
-        let mutablePost = this.get(id);
+        let mutablePost = Object.assign(this.get(id));
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: If `this.get(id)` will return null, you'll get an error here 😟

File: task3/UI/script_task7.js [hunk 7]
```diff
@@ -84,9 +84,8 @@ class AdCollection {
                     }
                 }
             )
-            if (AdCollection._validate(mutablePost) && this.get(adItem.id)) {
-                this.remove(id);
-                this.add(mutablePost);
+            if (AdCollection._validate(mutablePost) && this.get(id)) {
+                this._adList.splice(id, mutablePost);
```
Comment (YauheniyaBasaranovich) [inline task3/UI/script_task7.js]: Please take a look at `splice` docs: first argument is an index, second is amount and third is the new item to insert 🙂

File: task3/UI/script_task7.js (hunk 1 above)
Comment (YauheniyaBasaranovich) [reply task3/UI/script_task7.js]: You can, but first please fix a couple of new defects of `edit` method 🙂

File: task3/UI/script_task7.js (hunk 6 above)
Comment (Darina007) [reply task3/UI/script_task7.js]: Thanks!
I keep forgetting about such cases

Comment (YauheniyaBasaranovich) [reply task3/UI/script_task7.js]: That's OK, the more you practice the less you forget such things 🙂 
Now you can merge the PR.

**Pull request merged**